src/parser/parser.ts

```diff
@@ -122,7 +125,14 @@ export class TemplateRoot {
         delete overrides.OrganizationFile;
         delete overrides.OrganizationFileContents;
 
-        const obj = yamlParseContentWithIncludes(normalizedContentsForParser, dirname) as ITemplate;
+        let obj;
+        if (Path.extname(filename) === NUNJUCKS_FILE_EXTENSION) {
```

Comment:
> instead of looking at the file extension i would branche based on whether there is a TextTemplatingContext object.

I'm not sure I understand this.  We now have CFN files (.yml) and we have nunjucks files (.nj).  If we only look at  TextTemplatingContext how would we know which files nunjucks should parse? 

>I also think its makes sense to perform text-templating on the output of the parser (not mix those two)...

hmm, this request is not clear to me.  Are you saying that users should not mix CFN files with nunjucks files and the files should live in completely separate directories?

Reply:
My assumption is that it only makes sense to run a file through nunjucks if there is a TextTemplatingContext. therefore => if there is no TextTemplatingContext you dont run it through nunjucks.

I do think the file extension is a great convention, surely makes it easier for IDEs to interpret the file, but i wonder whether the branching logic in the cli should depend on it.

The last suggestion was to split out the parser from the TextTemplating. looking at the code perhaps the comment was not really thought through...

Reply:
Actually there are use cases where you may want to parse nunjucks files even if no Context data is passed in.  Here are a few use cases:
1. Nunjucks allows setting variables in templates therefore a user may set all of the  context data in the nunjucks template itself.  You can do this with a nunjucks [set](https://mozilla.github.io/nunjucks/templating.html#set)
2. Users may generate CFN from nunjucks template without needing to set any data.  For example, lets say I want to create N number of the same buckets.  I can just create a nunjucks template with one `AWS::S3::Bucket` then loop N times with a nunjucks [range function](https://mozilla.github.io/nunjucks/templating.html#range-start-stop-step).

Reply:
would those work if we inject {} as Context? that way `TextTemplatingContext: {}` could still be indicative of having to run text-templating

Reply:
I'm not sure that's better than using the filename as an indication to parse.  I'm also not a fan of making users create an empty object in their template, it's not real intuitive that you need to do that to enable templating.  With that being said I'm also ok with this approach if that's what you prefer.

Reply:
great, thanks. lets start out this way if there is feedback on this we can always change.

Reply:
refactored to enable templating only if a `TextTemplatingContext` in ofn task file is set.